include/mrdox/Plugin.hpp

```diff
@@ -0,0 +1,45 @@
+//
+// This is a derivative work. originally part of the LLVM Project.
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// Copyright (c) 2023 Vinnie Falco (vinnie.falco@gmail.com)
+//
+// Official repository: https://github.com/cppalliance/mrdox
+//
+#ifndef MRDOX_PLUGIN_HPP
+#define MRDOX_PLUGIN_HPP
+
+#include <mrdox/Platform.hpp>
+#include <mrdox/Support/Error.hpp>
+
+namespace clang {
+namespace mrdox {
+
+class Generator;
+class PluginEnvironment
+{
+public:
+    virtual
+    void
+    reportError(Error error) = 0;
```

Comment:
Because the tool puts them into a warning. Because a plugin failure isn't critical. You reckon I should just remove this and let the plugin use `report::error` as it sees fit?

Reply:
Yes that makes more sense